From aa967c93994e98ad22c4c8f12f4a4be447649545 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Wed, 27 May 2020 21:00:06 +0530
Subject: [PATCH 10/10] otg-peripheral-and-host-mode-support

- This patch will add support to OTG peripheral mode and adds separate dts file for OTG host mode support.
- Removes ptn5110 usb typec connector and ptn36043 usb typec SuperSpeed active switch support.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm64/boot/dts/adlink/Makefile           |  5 +-
 ...ink-lec-imx8m-1gb-ddr3l-usb-otg-master.dts | 12 +++
 ...ink-lec-imx8m-2gb-ddr3l-usb-otg-master.dts | 12 +++
 .../adlink-lec-imx8m-usb-otg-master.dts       | 12 +++
 .../boot/dts/adlink/adlink-lec-imx8m.dts      | 77 +------------------
 arch/arm64/configs/lec_imx8m_defconfig        |  4 -
 6 files changed, 41 insertions(+), 81 deletions(-)
 create mode 100644 arch/arm64/boot/dts/adlink/adlink-lec-imx8m-1gb-ddr3l-usb-otg-master.dts
 create mode 100644 arch/arm64/boot/dts/adlink/adlink-lec-imx8m-2gb-ddr3l-usb-otg-master.dts
 create mode 100644 arch/arm64/boot/dts/adlink/adlink-lec-imx8m-usb-otg-master.dts

diff --git a/arch/arm64/boot/dts/adlink/Makefile b/arch/arm64/boot/dts/adlink/Makefile
index 8a6bd43cd78e..67d4b0bef837 100644
--- a/arch/arm64/boot/dts/adlink/Makefile
+++ b/arch/arm64/boot/dts/adlink/Makefile
@@ -1,5 +1,8 @@
 # SPDX-License-Identifier: GPL-2.0
 
 dtb-$(CONFIG_ARCH_LEC_IMX8M) += adlink-lec-imx8m.dtb \
+				adlink-lec-imx8m-usb-otg-master.dtb \
 				adlink-lec-imx8m-2gb-ddr3l.dtb \
-				adlink-lec-imx8m-1gb-ddr3l.dtb
+				adlink-lec-imx8m-2gb-ddr3l-usb-otg-master.dtb \
+				adlink-lec-imx8m-1gb-ddr3l.dtb \
+				adlink-lec-imx8m-1gb-ddr3l-usb-otg-master.dtb
diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-1gb-ddr3l-usb-otg-master.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-1gb-ddr3l-usb-otg-master.dts
new file mode 100644
index 000000000000..a5dc9f3b590c
--- /dev/null
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-1gb-ddr3l-usb-otg-master.dts
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright 2017 NXP
+ * Copyright (C) 2017-2018 Pengutronix, Lucas Stach <kernel@pengutronix.de>
+ */
+
+#include "adlink-lec-imx8m-1gb-ddr3l.dts"
+
+&usb_dwc3_0 {
+        dr_mode = "host";
+        status = "okay";
+};
diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-2gb-ddr3l-usb-otg-master.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-2gb-ddr3l-usb-otg-master.dts
new file mode 100644
index 000000000000..a05df4b13b48
--- /dev/null
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-2gb-ddr3l-usb-otg-master.dts
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright 2017 NXP
+ * Copyright (C) 2017-2018 Pengutronix, Lucas Stach <kernel@pengutronix.de>
+ */
+
+#include "adlink-lec-imx8m-2gb-ddr3l.dts"
+
+&usb_dwc3_0 {
+        dr_mode = "host";
+        status = "okay";
+};
diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-usb-otg-master.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-usb-otg-master.dts
new file mode 100644
index 000000000000..d78f66f1f6d3
--- /dev/null
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m-usb-otg-master.dts
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright 2017 NXP
+ * Copyright (C) 2017-2018 Pengutronix, Lucas Stach <kernel@pengutronix.de>
+ */
+
+#include "adlink-lec-imx8m.dts"
+
+&usb_dwc3_0 {
+        dr_mode = "host";
+        status = "okay";
+};
diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
index 99de50e65403..08f84161cdd3 100755
--- a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
@@ -63,20 +63,6 @@
 		clock-frequency = <100000000>;
 	};
 
-	ptn36043 {
-		compatible = "nxp,ptn36043";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_ss_sel>;
-		switch-gpios = <&gpio3 15 GPIO_ACTIVE_HIGH>;
-		orientation-switch;
-
-		port {
-			usb3_data_ss: endpoint {
-				remote-endpoint = <&typec_con_ss>;
-			};
-		};
-	};
-
 	reg_usdhc2_vmmc: regulator-vsd-3v3 {
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_reg_usdhc2>;
@@ -441,45 +427,6 @@
 			};
 		};
 	};
-
-	ptn5110: tcpc@50 {
-		compatible = "nxp,ptn5110";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_typec>;
-		reg = <0x50>;
-		interrupt-parent = <&gpio3>;
-		interrupts = <3 8>;
-
-		port {
-			typec_dr_sw: endpoint {
-				remote-endpoint = <&usb3_drd_sw>;
-			};
-		};
-
-		usb_con: connector {
-			compatible = "usb-c-connector";
-			label = "USB-C";
-			power-role = "dual";
-			try-power-role = "sink";
-			source-pdos = <PDO_FIXED(5000, 3000, PDO_FIXED_USB_COMM)>;
-			sink-pdos = <PDO_FIXED(5000, 3000, PDO_FIXED_USB_COMM)
-				     PDO_VAR(5000, 20000, 3000)>;
-			op-sink-microwatt = <15000000>;
-			self-powered;
-
-			ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				port@1 {
-					reg = <1>;
-					typec_con_ss: endpoint {
-						remote-endpoint = <&usb3_data_ss>;
-					};
-				};
-			};
-		};
-	};
 };
 
 &i2c3 {
@@ -632,18 +579,8 @@
 };
 
 &usb_dwc3_0 {
-	dr_mode = "otg";
-	hnp-disable;
-	srp-disable;
-	adp-disable;
-	usb-role-switch;
+	dr_mode = "peripheral";
 	status = "okay";
-
-	port {
-		usb3_drd_sw: endpoint {
-			remote-endpoint = <&typec_dr_sw>;
-		};
-	};
 };
 
 &usb3_phy1 {
@@ -820,18 +757,6 @@
 		>;
 	};
 
-	pinctrl_ss_sel: usb3ssgrp{
-		fsl,pins = <
-			MX8MQ_IOMUXC_NAND_RE_B_GPIO3_IO15		0x16
-		>;
-	};
-
-	pinctrl_typec: typecgrp {
-		fsl,pins = <
-			MX8MQ_IOMUXC_NAND_CE2_B_GPIO3_IO3		0x17059
-		>;
-	};
-
 	pinctrl_uart1: uart1grp {
 		fsl,pins = <
 			MX8MQ_IOMUXC_UART1_RXD_UART1_DCE_RX		0x49
diff --git a/arch/arm64/configs/lec_imx8m_defconfig b/arch/arm64/configs/lec_imx8m_defconfig
index 1bcdf7ebb153..5b051ad51271 100644
--- a/arch/arm64/configs/lec_imx8m_defconfig
+++ b/arch/arm64/configs/lec_imx8m_defconfig
@@ -584,10 +584,6 @@ CONFIG_USB_AUDIO=m
 CONFIG_USB_ETH=m
 CONFIG_USB_MASS_STORAGE=m
 CONFIG_USB_G_SERIAL=m
-CONFIG_TYPEC=y
-CONFIG_TYPEC_TCPM=y
-CONFIG_TYPEC_TCPCI=y
-CONFIG_TYPEC_SWITCH_GPIO=y
 CONFIG_MMC=y
 CONFIG_MMC_BLOCK_MINORS=32
 CONFIG_MMC_ARMMMCI=y
-- 
2.17.1

